### ISSUE TITLE
Support LDAP authentication

### ISSUE THREAD

**Issue by lfrancke** [stackabletech/druid-operator#144]
As a user I'd like to use my existing [LDAP/AD](https://druid.apache.org/docs/latest/operations/auth-ldap.html) credentials to log into Druid. This was already done in e.g. [NiFi](https://github.com/stackabletech/nifi-operator/pull/303) or [Trino](https://github.com/stackabletech/trino-operator/pull/306). This can be especially helpful for writing tests. 

The LDAP support should be integrated in the structure from PR https://github.com/stackabletech/druid-operator/issues/6 which must be finished first. 
```
apiVersion: druid.stackable.tech/v1alpha1
kind: DruidCluster
metadata:
  name: druid
spec:
  version: 24.0.0-stackable0.1.0
  clusterConfig:
    tls:
      serverSecretClass: String # defaults to "tls"
      internalSecretClass: String # defaults to "tls"
    authentication: 
      - authenticationClass: druid-tls-authentication-class # String
      - authenticationClass: druid-ldap-authentication-class # String
```

This is done when
- [ ] LDAP is configurable in the CRD using the [LDAP structs](https://github.com/stackabletech/operator-rs/blob/main/src/commons/ldap.rs) from operator-rs
- [ ] A user/admin can configure Druid to use a LDAP server for authentication (while still offering existing authentication methods)
- [ ] There is documentation on how to configure Druid with LDAP using the Custom Resource
- [ ] Optional: There is an example demonstrating Druid with LDAP (docs, or example folder)
- There are tests which include:
  - [ ] OpenLDAP is installed and accessible via Druid
  - [ ] LDAP authenticated access to Druid works
- [ ] It is added to the feature tracker (ask Lars for help)

This depends on the reference architecture developed in https://github.com/stackabletech/issues/issues/170

**Comment by fhennig** [comment]
Druid already supports LDAP authentication https://druid.apache.org/docs/latest/operations/auth-ldap.html

**Comment by lfrancke** [comment]
Blocked by https://github.com/stackabletech/druid-operator/issues/6

**Comment by razvan** [comment]
This ticket needs some refinement because it's not enough to enable LDAP authentication to access Druid. It's also required to add users to Druid using [coordinator REST API](https://druid.apache.org/docs/latest/operations/security-overview.html#enable-authorizers).

**Comment by sbernauer** [comment]
Blocked by https://github.com/stackabletech/druid-operator/issues/365

**Comment by vsupalov** [comment]
The first PR https://github.com/stackabletech/druid-operator/pull/374 is in review now. To close this ticket, we'll need to add ldaps:// in a later PR. They are separated for the sake of reviewer convenience, as the second part will require refactoring TLS-related code as well.

**Comment by vsupalov** [comment]
Follow-up tickets from the initial [LDAP feature implementation PR](https://github.com/stackabletech/druid-operator/pull/374), to be prioritized:

* https://github.com/stackabletech/druid-operator/issues/382
* https://github.com/stackabletech/druid-operator/issues/383
* https://github.com/stackabletech/druid-operator/issues/384

**Comment by lfrancke** [comment]
I briefly talked to Vladislav today and he said that we might want to do more work here before we finish this ticket.
Can someone give an update on this issue?

**Comment by vsupalov** [comment]
My current understanding: there is the intention to add ldaps support to this ticket before closing it, as it's not supposed to be much effort and can be considered important functionality.

The previous PR was very long-running and already quite big, which is why I opted to close it instead of cramming #382 into it as well. It might be a relatively simple PR, but from what I could tell a lot of existing TLS functionality would need to be adjusted as well (creating and handling a common trust store mostly).
